### PR TITLE
modify log => klog

### PR DESCRIPTION
**Pull request by MrLeea-13155bc**
**What this PR does / why we need it**:
using klog instead of log
**Will this PR make the community happier**? 
yes
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: examples/opcuaDeviceShifu/server/generate_cert.go [hunk 1]
```diff
@@ -74,39 +75,39 @@ func main() {
...
 	keyOut, err := os.OpenFile(keyFile, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
 	if err != nil {
-		log.Printf("failed to open %s for writing: %s", keyFile, err)
+		klog.Errorf("failed to open %s for writing: %s", keyFile, err)
 		return
 	}
 
 	if err := pem.Encode(keyOut, pemBlockForKey(priv)); err != nil {
-		log.Fatalf("failed to write data to %s: %s", keyFile, err)
+		klog.Errorf("failed to write data to %s: %s", keyFile, err)
```
Comment (BtXin) [inline examples/opcuaDeviceShifu/server/generate_cert.go:103]: Why Fatalf -> Errorf?

File: pkg/deviceshifu/deviceshifubase/deviceshifubase.go [hunk 2]
```diff
@@ -127,13 +127,13 @@ func deviceHealthHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func instructionNotFoundHandler(w http.ResponseWriter, r *http.Request) {
-	log.Printf("Error: Device instruction does not exist!")
+	klog.Infof("Error: Device instruction does not exist!")
```
Comment (BtXin) [inline pkg/deviceshifu/deviceshifubase/deviceshifubase.go:130]: Infof->Error

File: pkg/deviceshifu/deviceshifubase/deviceshifubase.go [hunk 3]
```diff
@@ -251,7 +251,7 @@ func (ds *DeviceShifuBase) StartTelemetryCollection(fn collectTelemetry) error {
 	for {
 		err := ds.telemetryCollection(fn)
 		if err != nil {
-			log.Println("error when telemetry collection")
+			klog.Errorln("error when telemetry collection")
```
Comment (BtXin) [inline pkg/deviceshifu/deviceshifubase/deviceshifubase.go:254]: Log the err itself as well

File: pkg/deviceshifu/deviceshifubase/deviceshifubase.go [hunk 4]
```diff
@@ -265,18 +265,18 @@ func (ds *DeviceShifuBase) startHTTPServer(stopCh <-chan struct{}) error {
...
 // Start HTTP server and telemetryCollection
 func (ds *DeviceShifuBase) Start(stopCh <-chan struct{}, fn collectTelemetry) error {
-	log.Printf("deviceshifu %s started\n", ds.Name)
+	klog.Infof("deviceshifu %s started\n", ds.Name)
 
 	go func() {
 		err := ds.startHTTPServer(stopCh)
 		if err != nil {
-			log.Println("error during Http Server is up, error: ", err)
+			klog.Errorln("error during Http Server is up, error: ", err)
```
Comment (BtXin) [inline pkg/deviceshifu/deviceshifubase/deviceshifubase.go:273]: Shouldn't this be Errorf?

File: pkg/deviceshifu/deviceshifubase/deviceshifubase.go [hunk 5]
```diff
@@ -288,6 +288,6 @@ func (ds *DeviceShifuBase) Stop() error {
 		return err
 	}
 
-	log.Printf("deviceshifu %s's http server stopped\n", ds.Name)
+	klog.Infof("deviceshifu %s's http server stopped\n", ds.Name)
```
Comment (BtXin) [inline pkg/deviceshifu/deviceshifubase/deviceshifubase.go:291]: Don't need to add \n, klog will add this for you. Same for every other places

File: pkg/deviceshifu/deviceshifubase/deviceshifubase_config.go [hunk 6]
```diff
@@ -119,15 +119,15 @@ func NewDeviceShifuConfig(path string) (*DeviceShifuConfig, error) {
 	if instructions, ok := cfg[ConfigmapInstructionsStr]; ok {
 		err := yaml.Unmarshal([]byte(instructions), &dsc.Instructions)
 		if err != nil {
-			log.Fatalf("Error parsing %v from ConfigMap, error: %v", ConfigmapInstructionsStr, err)
+			klog.Fatalf("Error parsing %v from ConfigMap, error: %v", ConfigmapInstructionsStr, err)
```
Comment (BtXin) [inline pkg/deviceshifu/deviceshifubase/deviceshifubase_config.go:122]: Do you still need to return err if you fatal here? Or maybe we should avoid using fatal and return error to the callee at every call.

File: pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go [hunk 7]
```diff
@@ -98,7 +98,7 @@ func New(deviceShifuMetadata *deviceshifubase.DeviceShifuMetaData) (*DeviceShifu
 				mux.HandleFunc("/"+instruction, handler.commandHandleFunc())
 			}
 		default:
-			log.Printf("EdgeDevice protocol %v not supported in deviceShifu_http_http\n", protocol)
+			klog.Warningf("EdgeDevice protocol %v not supported in deviceShifu_http_http\n", protocol)
```
Comment (BtXin) [inline pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go:101]: If it's just a warning you should not return error. Use klog.Errorf here.

File: pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go [hunk 8]
```diff
@@ -207,20 +207,20 @@ func (handler DeviceCommandHandlerHTTP) commandHandleFunc() http.HandlerFunc {
...
 			if httpErr != nil {
 				http.Error(w, httpErr.Error(), http.StatusServiceUnavailable)
-				log.Printf("HTTP POST error" + httpErr.Error())
+				klog.Errorf("HTTP POST error" + httpErr.Error())
 				return
 			}
 		default:
 			http.Error(w, httpErr.Error(), http.StatusBadRequest)
-			log.Println("Request type " + reqType + " is not supported yet!")
+			klog.Warningln("Request type " + reqType + " is not supported yet!")
```
Comment (BtXin) [inline pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go:223]: Same here

File: pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go [hunk 9]
```diff
@@ -229,16 +229,16 @@ func (handler DeviceCommandHandlerHTTP) commandHandleFunc() http.HandlerFunc {
...
 			if err != nil {
-				log.Println("error when copy requestBody from responseBody, err: ", err)
+				klog.Errorln("error when copy requestBody from responseBody, err: ", err)
 			}
 			return
 		}
 
 		// TODO: For now, just write tht instruction to the response
-		log.Println("resp is nil")
+		klog.Infoln("resp is nil")
```
Comment (BtXin) [inline pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go:238]: You can use warning here.

File: pkg/deviceshifu/deviceshifuopcua/deviceshifuopcua.go [hunk 10]
```diff
@@ -221,16 +221,16 @@ func (ds *DeviceShifu) requestOPCUANodeID(nodeID string) error {
...
 	}
 
 	if resp.Results[0].Status != ua.StatusOK {
-		log.Printf("OPC UA response status is not OK, status: %v", resp.Results[0].Status)
+		klog.Errorf("OPC UA response status is not OK, status: %v", resp.Results[0].Status)
 		return err
 	}
 
-	log.Println(resp.Results[0].Value.Value())
+	klog.Infoln(resp.Results[0].Value.Value())
```
Comment (BtXin) [inline pkg/deviceshifu/deviceshifuopcua/deviceshifuopcua.go:233]: You used klog.Infoln here, and used klog.Infof at line 187 with exactly the same content. Please make the log consistent, don't use two different methods for same content. In fact, just use Infof, Errorf... etc for all logs. klog will and newline for you.

**Review by BtXin — CHANGES_REQUESTED** [review]

File: pkg/deviceshifu/deviceshifubase/deviceshifubase_config.go (hunk 6 above)
Comment (seanli33056) [reply pkg/deviceshifu/deviceshifubase/deviceshifubase_config.go:122]: bad configmap should be a fatal

File: pkg/deviceshifu/deviceshifuopcua/deviceshifuopcua.go (hunk 10 above)
Comment (saiyan86) [reply pkg/deviceshifu/deviceshifuopcua/deviceshifuopcua.go]: Why is this resolved? @MrLeea-13155bc Please don't resolve the comment before addressing it.

Comment (saiyan86) [reply pkg/deviceshifu/deviceshifuopcua/deviceshifuopcua.go]: I still see a bunch of ```...ln(...)```

Comment (MrLeea-13155bc) [reply pkg/deviceshifu/deviceshifuopcua/deviceshifuopcua.go]: forget to push  😢

File: examples/opcuaDeviceShifu/server/generate_cert.go [hunk 11]
```diff
@@ -90,7 +90,7 @@ func main() {
 		log.Fatalf("error closing %s: %s", certFile, err)
 	}
 
-	log.Printf("wrote %s\n", certFile)
+	log.Printf("wrote %s", certFile)
```
Comment (BtXin) [inline examples/opcuaDeviceShifu/server/generate_cert.go:93]: let's keep things consistent, for all logging places, let's use klog.

File: pkg/deviceshifu/deviceshifubase/deviceshifubase.go [hunk 12]
```diff
@@ -127,13 +127,13 @@ func deviceHealthHandler(w http.ResponseWriter, r *http.Request) {
...
 func instructionNotFoundHandler(w http.ResponseWriter, r *http.Request) {
-	log.Printf("Error: Device instruction does not exist!")
+	klog.Errorf("Error: Device instruction does not exist!")
 	http.Error(w, "Error: Device instruction does not exist!", http.StatusNotFound)
 }
 
 // UpdateEdgeDeviceResourcePhase Update device status
 func (ds *DeviceShifuBase) UpdateEdgeDeviceResourcePhase(edPhase v1alpha1.EdgeDevicePhase) {
-	log.Printf("updating device %v status to: %v\n", ds.Name, edPhase)
+	klog.Infof("updating device %v status to: %v\n", ds.Name, edPhase)
```
Comment (BtXin) [inline pkg/deviceshifu/deviceshifubase/deviceshifubase.go:136]: remove the "\n" here and everywhere else.

File: pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go [hunk 13]
```diff
@@ -295,37 +295,37 @@ func (handler DeviceCommandHandlerHTTPCommandline) commandHandleFunc() http.Hand
...
 			if err != nil {
-				log.Println("cannot copy requestBody from requestBody, error: ", err)
+				klog.Errorf("cannot copy requestBody from requestBody, error: %v", err)
 			}
 			return
 		}
 
 		// TODO: For now, just write tht instruction to the response
-		log.Println("resp is nil")
+		klog.Infof("resp is nil")
```
Comment (BtXin) [inline pkg/deviceshifu/deviceshifuhttp/deviceshifuhttp.go:325]: Use Warningf

File: tools/tag.go [hunk 14]
```diff
@@ -93,7 +94,7 @@ func replaceTag(file *os.File, oldTag string, newTag string) error {
 				return nil
 			}
 
-			log.Printf("cannot read file %s %v", file.Name(), err)
+			klog.Infof("cannot read file %s %v", file.Name(), err)
```
Comment (BtXin) [inline tools/tag.go:97]: use Errorf

File: examples/opcuaDeviceShifu/server/generate_cert.go (hunk 11 above)
Comment (MrLeea-13155bc) [reply examples/opcuaDeviceShifu/server/generate_cert.go:93]: this file not run in deviceshifu,  just to generate certification for opcua, user can use `go run generate_cert.go
` directly

Comment (BtXin) [reply examples/opcuaDeviceShifu/server/generate_cert.go:93]: Yeah, but they are still in the same repo and share the same go.mod, so the logging should be consistent.

Comment (MrLeea-13155bc) [reply examples/opcuaDeviceShifu/server/generate_cert.go:93]: this file just a tool for generate certificate, it shouldn't rely on go.mod, using klog could make things more complicated.

Comment (BtXin) [reply examples/opcuaDeviceShifu/server/generate_cert.go:93]: use fmt.Print instead then.

File: tools/tag.go (hunk 14 above)
Comment (BtXin) [reply tools/tag.go]: use fmt.Print intead

**Review by BtXin — APPROVED** [review]

**Review by BtXin — CHANGES_REQUESTED** [review]
Change the place you used log.Fatal(not klog.Fatal) to panic

**Comment by MrLeea-13155bc** [comment]
> Change the places you used log.Fatal(not klog.Fatal) to panic

Why

**Comment by BtXin** [comment]
> 

Because you are still using log here.

**Comment by MrLeea-13155bc** [comment]
> > 
> 
> 
> 
> Because you are still using log here. 

so what？

**Comment by BtXin** [comment]
> > > 
> > 
> > 
> > Because you are still using log here.
> 
> so what？

So if you are using log, please use a proper logger like klog, otherwise don't use it.

**Comment by BtXin** [comment]
1. It is generally a bad practice to mix use of print and log
2. I believe print out the full stacktrace for tools like tag and generate Cert would help both users and developers to debug and develop.

**Review by BtXin — APPROVED** [review]

**Pull request merged**